### PR TITLE
65 research n publications

### PR DESCRIPTION
**Pull request by naina66406**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--world-bank--aemsites.hlx.live/
- After: https://65-research-n-publications--world-bank--aemsites.hlx.live/en/drafts/naina/research-and-publication

### CODE REVIEW

File: blocks/research-publications-cards/research-publications-cards.css [hunk 1]
```diff
@@ -0,0 +1,158 @@
+.title-button-wrapper{
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 50px;
+
+    .title{
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.css:7]: Can you format your entire CSS file once. Indentation and bracket spacing looks off.

File: blocks/research-publications-cards/research-publications-cards.css [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        p{
+            margin: 0;
+        }
+    }
+    
+    a.title-button{
+        background: var(--primary-blue-80);
+        padding: 15px 20px;
+        border-radius: 50px;
+        font-size: var(--size-16);
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.css:21]: Ideally this should be picked up from font variables. Only use this if the value is not available there.

File: blocks/research-publications-cards/research-publications-cards.css [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    .rp-card{
+        display: grid;
+        grid-template-columns: auto auto;
+        gap: 40px;
+
+        .text-content{
+            .title{
+                font-size: var(--size-18);
+                line-height: 22px;
+                color: #004370;
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.css:63]: Check if variable is available.

File: blocks/research-publications-cards/research-publications-cards.css [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                padding: 5px 0 15px;
+                border-top: 1px solid #E7EDF3;
+                margin-top: 40px;
+            }
+
+            a{
+                text-decoration: none;
+
+                .card-btn{
+                    font-size: var(--size-16);
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.css:83]: See if you can use one of the font variables.
```css
/* typography specific - h1..h6, body ..etc */
    --h1-huge: var(--size-60);
    --h1-large: var(--size-44);
    --h1-base: var(--size-36);
    --h2-large: var(--size-32);
    --h2-base: var(--size-32);
    --h3-large: var(--size-28);
    --h3-base: var(--size-28);
    --heading-label-large: var(--size-36);
    --heading-label-base: var(--size-28);
    --heading-label-small: var(--size-28);
    --heading-label-tiny: var(--size-24);

    /* Body variables */
    --default: var(--size-20);
    --body-huge: var(--size-20);
    --body-large: var(--size-18);
    --body-medium: var(--size-14);
    --body-small: var(--size-14);
    --body-tiny: var(--size-12);
```
These scale based on viewport.

File: blocks/research-publications-cards/research-publications-cards.css [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+    .cards-container .rp-card .text-content .title{
+        font-size: var(--size-24);
+        line-height: 29px;
+
+    }
+            
+}
+
+@media(width >=900){
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.css:151]: We don't use this breakpoint value. Can we correct this?

File: blocks/research-publications-cards/research-publications-cards.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    btnLink,
+  ] = [...cardData.children];
+  const imgElement = imageContainer.querySelector('img');
+  imgElement.setAttribute('alt', imageAltText.textContent);
+  imageContainer.className = 'card-img';
+  title.className = 'title';
+  desc.className = 'desc';
+  btn.className = 'card-btn';
+
+  const RPCard = div({ class: 'rp-card' }, imageContainer, div({ class: 'text-content' }, a({ href: titlelink.textContent }, title), desc, a({ href: btnLink.textContent }, btn)));
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.js:23]: Please don't capitalize variables unless it is a constructor function in JS.

File: blocks/research-publications-cards/research-publications-cards.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  titlelink.remove();
+  btnLink.remove();
+  imageAltText.remove();
+  if (!cardsContainer) {
+    cardsContainer = div({ class: 'cards-container' }, RPCard);
+  } else {
+    cardsContainer.append(RPCard);
+  }
+  document.querySelector('.research-publications-cards').append(cardsContainer);
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.js:32]: Is it possible to reduce scope here and query within this block?

File: blocks/research-publications-cards/research-publications-cards.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+export default async function decorate(block) {
+  const [title, button, link, rpFirstCard, rpSecondCard] = [...block.children];
```
Comment (thomsebastin) [inline blocks/research-publications-cards/research-publications-cards.js:36]: There can be `n` number of cards authored. Could you please handle that in your JS? 
```js
const [title, button, link, ...rpCards] = [...block.children];
```
like so..

**Review by thomsebastin — CHANGES_REQUESTED** [review]

**Review by mkbansal1 — APPROVED** [review]

**Review by thomsebastin — APPROVED** [review]

**Pull request merged**